nixos/modules/virtualisation/containers.nix

```diff
@@ -28,6 +29,26 @@ in
       description = lib.mdDoc "Enable the OCI seccomp BPF hook";
     };
 
+    cdi = mkOption {
+      type = types.attrsOf json.type;
+      default = { };
+      description = lib.mdDoc ''
+        This option configures the Container Device Interface (CDI) of the system. Every key of the attribute set will be mapped to a file in /etc/cdi, and is required to be provided in JSON format.
+      '';
+      example = literalExpression ''
+        {
+          nvidia = builtins.fromJSON '''
+            {
+              "cdiVersion": "0.5.0",
+              "kind": "nvidia.com/gpu",
+              "devices": [ ... ],
+              "containerEdits": [ ... ]
+            }
+          ''';
+        }
```

Comment:
Might as well just write it down as an attrset, that's probably cleaner?

Reply:
I would like to avoid this, tbh. Unless I'm misunderstanding I can totally see a trend of vendors providing tools to auto generate CDI specs based on the detected hardware. These tools will output JSON directly, and I believe is a bit of a burden having to ask people to translate that to attrsets manually.

Unless I'm missing something, I think we are better served by just making sure that the JSON provided is a syntactically valid JSON.

Reply:
H'm? The value of `config.virtualisation.containers.cdi.${name}` is going to be an attrset, as the value produced by `fromJSON`. JSON is embeds into our attrsets. It's a feature that we model this value as an attrset and not a json string, because this way we can compose it from a bunch of definitions using the module system. It's probably a good idea to add the hint that users may copy-paste JSON snippets as is and use `fromJSON`. I think it might be misleading to put it in the `example` field though

Reply:
Oh, you are right. I'll update the PR.